crates/matrix-sdk/src/event_cache/store.rs

```diff
@@ -130,6 +166,126 @@ impl RoomEvents {
     pub fn events(&self) -> impl Iterator<Item = (Position, &Event)> {
         self.chunks.items()
     }
+
+    /// Deduplicate `events` considering all events in `Self::chunks`.
+    ///
+    /// The returned tuple contains (i) the unique events, and (ii) the
+    /// duplicated events (by ID).
+    fn filter_duplicated_events<'a, I>(&'a mut self, events: I) -> (Vec<Event>, Vec<OwnedEventId>)
+    where
+        I: Iterator<Item = Event> + 'a,
+    {
+        let mut duplicated_event_ids = Vec::new();
+
+        let deduplicated_events = self
+            .deduplicator
+            .scan_and_learn(events, self)
+            .filter_map(|decorated_event| match decorated_event {
+                Decoration::Unique(event) => Some(event),
+                Decoration::Duplicated(event) => {
+                    error!(event_id = ?event.event_id(), "Found a duplicated event");
```

Comment:
not an error, it's pretty benign, so `trace` or `debug` please :)

Reply:
Went for a `debug!`.